#Main Dimensions of Stator Core

\begin{equation}
  D_{is} = \sqrt[3]{\frac{2p*p_1*S_{gap}}{pi*\lambda*f_1*C_0}}
\end{equation}

In [1]:
P_n = 150 
V_1 = 1000
n_1 = 1800 
p   = 2
p_1 = 2
f_1 = 50
cos_theta = 0.85
K_E = 0.98 - 0.005*p_1
eff = 0.9
q=3


\begin{equation}
  S_{gap} = \frac{K_E*P_n}{\eta*\cos(\theta_n)}
\end{equation}

In [2]:
print K_E

0.97


In [3]:
S_gap = ((K_E*150*1000)/(eff*cos_theta))
print S_gap

190196.078431


###for a 4 pole machine,
\begin{equation}
\lambda = 1.1
\end{equation}

In [4]:
import math
D_is = math.pow((2*p*p_1*S_gap)/(3.14*1.1*f_1*265000),1.0/3.0)
print D_is

0.321551923037


\begin{equation}
D_{is} = 0.32 m
\end{equation}

\begin{equation}
\tau = \frac{pi*D_{is}}{2*p}
\\\tau_{s} = \frac{\tau}{3*q}
\end{equation}

In [5]:
pole_pitch = 3.14*D_is/(2*p)
print pole_pitch

0.252418259584


\begin{equation}
\tau = 0.25 m
\end{equation}

In [6]:
slot_per_pole= pole_pitch/(3*q)
print slot_per_pole

0.0280464732871


####again for a 4 pole machine;
\begin{equation}
\frac{D_{is}}{D_{out}} = K_D = 0.62
\end{equation}

In [7]:
D_out = D_is/0.62
print D_out

0.51863213393


In [8]:
D_out=0.53


\begin{equation}
D_{out} = 0.53 m
\end{equation}

\begin{equation}
g = (0.1 + 0.012*\sqrt[3]{P_n}*10^{-3})
\\for \quad 2p>2
\end{equation}

In [9]:
import math
g = (0.1+0.012*math.pow(P_n*1000,1.0/3.0))/(1000)
print g

0.00073759514151


\begin{equation}
g = 0.7 mm
\end{equation}

#The Stator Winding 

\begin{equation}
N_s = 2*p*q*m = 2*2*3*3 = 36
\end{equation}

\begin{equation}
\alpha_{ec} = \frac{2\pi*p}{N_s} = \frac{\pi}{9}
\end{equation}

\begin{equation}
K_{ql} = \frac{\sin(\frac{\pi}{6})}{q*\sin(\frac{\pi}{6q})}
\end{equation}

In [10]:
import math
K_ql = math.sin(3.14/6)/(3*math.sin(3.14/18))
print K_ql

0.959835420315


\begin{equation}
for\quad chording : \frac{y}{\tau} = \frac{7}{9}\quad is \quad selected
\end{equation}

\begin{equation}
so\quad the\quad electrical\quad angle\quad is:\quad \frac{2*\pi*p}{N_s}=\quad \frac{\pi}{9}
\end{equation}

\begin{equation}
chording\quad factor\quad is:\quad \sin(\frac{pi*y}{2*\tau})
\end{equation}

In [11]:
import math
K_yl = math.sin((3.14*7)/(2*9))
print K_yl

0.939480605157


In [12]:
import math
Kw1= K_yl*K_ql
print Kw1

0.901746761529


####following values are assumed from now on:


In [13]:
B_g = 0.7 
alpha_i = 0.65
L=1.1*3.14*D_is/(2*p)
rated_current = 265000/(1.1*alpha_i*Kw1*3.14*B_g*p)
print rated_current/1000


93.4969743854


In [14]:
pole_flux=alpha_i*pole_pitch*L*B_g
print pole_flux

0.0318893463746


####number of turns per phase W1:
\begin{equation}
\frac{K_E*V_1}{4*K_f*K_{W1}*f*\Phi}
\end{equation}

In [15]:
import math
W_1 = (K_E*V_1)/(4*1.1*Kw1*50*pole_flux*math.sqrt(3))
print W_1 

88.5234267262


####number of conductors ns:
\begin{equation}
ns:\quad \frac{W_1}{p*q}
\end{equation}
#### (current paths in parallel:1)

In [16]:
ns = W_1/(p*q)
print ns

14.7539044544


In [17]:
ns = 14

In [18]:
import math
rated_current= 150000/(eff*cos_theta*math.sqrt(3)*V_1)
print rated_current


113.205935135


In [19]:
J_co=6.3

####magnetic wire cross section:
\begin{equation}
A_{co}=\quad \frac{I_{rated}}{J_{cos}}
\end{equation}

In [20]:
A_co=rated_current/J_co
print A_co

17.9691960532


\begin{equation}
A_{co}=conductor\quad width*conductor\quad height
\end{equation}
####a rectengular cross section conductor will be used,so standart value of 10 mm is taken here for slot width

In [21]:
slot_width = 10
conductor_insulation = 0.4
slot_insulation = 4
conductor_width = slot_width - conductor_insulation - slot_insulation
print conductor_width

5.6


#### conductor height = A_co/conductor width = 18/5.6 = 3.21mm

In [148]:
radial_cond_insulation = conductor_insulation*ns
radial_slot_insulation = slot_insulation*2
interlayer_insulation = 2
wedge = 4
radial_insulation = radial_cond_insulation + radial_slot_insulation+interlayer_insulation+wedge
print radial_insulation

19.6


####slot height:
\begin{equation}
h_s = h_{ins} + n_s*b_c
\end{equation}

In [23]:
bc=3.21
h_s = radial_insulation + ns*3.21
print h_s

64.54


####back iron radial thickness:
\begin{equation}
h_{cs} = \frac{D_{out}-D_{is}}{2}-h_s
\end{equation}

In [24]:
h_cs= ((D_out-D_is)*1000)/2-h_s
print h_cs

39.6840384816


####so all in all;
\begin{equation}
h_s = 64.54 mm
\\h_{cs} = 39.7 mm
\end{equation}
####thus back iron flux density is:
\begin{equation}
B_{cs} = \frac{\Phi}{2*l_i*h_cs}
\end{equation}
####where l_i is stack length, and calculated as 0.277mm

In [25]:
B_cs = pole_flux/(2*L*0.001*h_cs)
print B_cs

1.44705922715


#### 1.8 T is a little big value, so D_out is changed to 0.53m and recalculations are made to obtain 1.44 T

#Rotor Slots

#### Ns was selected as 36 previously, so among the adequate combinations for 2p = 4 and Ns = 36, Nr, the skewed rotor slot number is chosen as 28

In [26]:
Nr=28

In [27]:
import math
Kw2 = math.sin(3.14/6)/(2.3*math.sin(3.14/14))
print Kw2

0.976985413207


\begin{equation}
\tau_r = \frac{\pi(D_{is}-2g)}{N_r}
\end{equation}

In [28]:
rotor_tooth_pitch = 3.14*(D_is*1000-2*g)/(Nr)
print rotor_tooth_pitch

36.0595859371


In [29]:
nr=2
W_2 = Nr*nr/(2*q*1)
print W_2

9


In [30]:
V_2 = V_1*(K_E*W_2*Kw2)/(W_1*Kw1)
print V_2

106.846306751


In [31]:
Ki= 0.936
import math
rated_current_2= (Ki*W_1*Kw1*rated_current)/(W_2*Kw2*math.sqrt(3))
print rated_current_2

555.388185846


In [32]:
j_cor = 7
A_cor = rated_current_2/(j_cor)
print A_cor

79.3411694066


####the slot width b_r is:
\begin{equation}
b_r<\tau_r(1-\frac{B_g}{B_{trmin}})
\end{equation}

In [33]:
Btrmin = (slot_per_pole*1000*B_g)/(slot_per_pole*1000-slot_width)
print Btrmin

1.08788742203


In [34]:
rotor_tooth_pitch*(1-B_g/Btrmin)

12.85708387216765

In [35]:
br=12

#### so 12 can be chosen as the slot width

####considering the insulation width is 2 mm, conductor allowable width is 8.5mm. For 2 vertical conductors,each conductor has 4mm of width. So the height is:

In [36]:
b_c = A_cor/(2*4)
print b_c

9.91764617582


####Rotor back iron height is:
\begin{equation}
h_{cr}=\frac{B_g*\tau*1}{B_{cr}*\pi*K_{Fe}}
\end{equation}


In [37]:
h_cr=(B_g*pole_pitch*1)/(1.4*3.14*0.98)
print h_cr

0.0410142758976


#Rotor performance calculation

####magnetization mmfs:

\begin{equation}
F_{lm}=F_g+F_s+(F_{st}+F_{rt})+\frac{F_{cs}}{2}+\frac{F_{cr}}{2}
\end{equation}

\begin{equation}
F_g=K_c*g*\frac{B_g}{\mu_o}
\\K_c=K_{c1}*K_{c2}
\\K_{c1,c2}=\frac{\tau_{s,r}}{\tau_{s,r}-\gamma_{s,r}g}
\\gamma_{s,r}=\frac{\frac{b_{os,r}}{g}^2}{5+\frac{b_{os,r}}{g}}
\end{equation}

In [38]:
import math
gamma=math.pow(slot_width,2)/(5*g*1000+slot_width)
print gamma


7.30568216488


In [39]:
Kc2= rotor_tooth_pitch/(rotor_tooth_pitch - gamma*g)
print Kc2

1.00014945932


In [40]:
Kc1=1.3

In [41]:
Kc=Kc1*Kc2
print Kc

1.30019429711


In [42]:
import math
Fg = (Kc*g*B_g)/(1.256*math.pow(10,-6))
print Fg

534.483994902


In [43]:
h_st=40
Ns=36
import math
Btsmin=math.pow(3.14*(D_is*1000+2*h_st*2)-Ns*slot_width,-1)*B_g*3.14*D_is*1000
print Btsmin

0.613477707851


In [44]:
h_rt=25
import math
Btrmax=3.14*(D_is*1000-2*g*1000)*B_g/(3.14*(D_is*1000-2*g*1000-2*h_rt)-Nr*br)
print Btrmax

1.37396954515


In [45]:
Btsmed=(3.14*D_is*1000*B_g)/((3.14*(D_is*1000+h_st)-Ns*slot_width))
print Btsmed

0.911641566115


In [46]:

Btrmed=(3.14*D_is*1000*B_g)/(3.14*(D_is*1000-h_rt)-Nr*br)
print Btrmed

1.18750528218


####from lamihation magnetization curve;
Hstmin=106 A/m, Hstmed = 177 A/m, Hstmax= 585 A/m
####and also;
\begin{equation}
F_{st}=h_{st}*H_{st}
\end{equation}

In [47]:
Hts=(106+4*177+585)/6
print Hts

233


In [48]:
Fst=h_st*0.001*Hts
print Fst

9.32


In [49]:
Btrmin = 1.15
Htrmin=310
Htrmed=356
Htrmax=760
Htr=(Htrmin+4*Htrmed+Htrmax)/6
print Htr

415


In [50]:
Frt=Htr*h_rt*0.001
print Frt

10.375


####Stator back core mmf Fcs is:
\begin{equation}
F_{cs}=l_{cs}*H_{cs}
\end{equation}

In [51]:
Bcsmax = (B_g*rotor_tooth_pitch*0.01)/(3.14*(D_out-D_is)/2)
print Bcsmax

0.771296268431


In [52]:
Hcsmax= 177

####the mean length lcs is:
\begin{equation}
l_{cs} = \frac{\pi*(D_{out}-h_{cs})}{2p}
\end{equation}

In [53]:
lcs=3.14*(D_out-h_cs*0.001)/(2*p)
print lcs


0.384898029792


In [54]:
Fcs=lcs*Hcsmax
print Fcs

68.1269512732


####same equations followed by rotor back iron mmf:

In [55]:
Hcr=237
D_ir=D_is-2*g*0.001-2*h_rt*0.001-2*h_cs*0.001
print D_ir
lcr=3.14*(D_ir-h_cr*0.001)*1.4/(2*p)
print lcr
Fcr=Hcr*lcr
print Fcr

0.192182370883
0.211163350912
50.045714166


####so the total magnetization mmf becomes:


In [56]:
Flm=Fg+(Frt+Fst)+Fcr/2+Fcs/2
print Flm

613.265327622


\begin{equation}
I_{\mu}=\frac{F_{lm}*p*\pi}{3\sqrt(2)*W_1*K_{w1}}
\end{equation}

In [57]:
import math
magnetization_current = Flm*p*3.14/(3*math.sqrt(2)*W_1*Kw1)
print magnetization_current

11.3717942758


In [58]:
current_ratio=magnetization_current/rated_current
print current_ratio

0.10045227984


#Rotor winding parameters

In [59]:
h_s1=ns*bc+ns*0.4+2*2+1
h_s3=2*2+1
h_s2=1*2+4
k_beta=(1+3*(7.0/9.0))/4
slot_leakage=(h_s1-h_s3)*k_beta/(3*slot_width)+h_s2*k_beta/(slot_width)+h_s3/(4*slot_width)
print slot_leakage

1.90388888889


In [60]:
import math
k01= 1 - 0.033*slot_width*slot_width/(g*1000*slot_per_pole*1000)
kapa01=0.003
differential_permeance=0.9*slot_per_pole*1000*q*Kw1*q*Kw1*k01*kapa01/(Kc*g*1000)
print differential_permeance

0.485684586194


In [61]:
l_i=0.277
l1=0.015
import math
lfs=2*(l1+(7.0/9.0)*pole_pitch/math.sin(0.52))+3.14*h_s*0.001
print lfs
end_permeance=0.34*(q/l_i)*(lfs-0.64*(7.0/9.0)*pole_pitch)
print end_permeance

1.02288768238
3.30391433823


In [62]:
X_sl=15.8*(f_1/100.0)*(W_1/100)*(W_1/100)*(l_i/(p*q))*(slot_leakage+differential_permeance+end_permeance)
print X_sl

1.62723560919


In [63]:
import math
K_80_deg=2.1712
Lgeo=l_i+6*0.0075
R_s=(W_1/A_co)*(Lgeo+lfs)*K_80_deg*math.pow(10,-2)
print R_s

0.143851838324


In [64]:
R_r=0.8*R_s
print R_r

0.115081470659


In [65]:
lfr=(pole_pitch-h_rt*0.001*3.14/4)*2/cos_theta+(4/3)*2*h_rt*0.001
print lfr

0.59774884608


In [66]:
Rr_r=(K_80_deg*2*(lfr+Lgeo)*W_2/A_cor)*0.01
print Rr_r

0.00453046719273


In [67]:
rotor_slot_leakage=(2*b_c)/(3*br)+1.2/(4*br)+1.2/br+0.5+wedge/(br+5)
print rotor_slot_leakage


1.1759803431


In [68]:
X_m=(V_1-magnetization_current*X_sl)/magnetization_current
print X_m

86.3096348395


#Losses and Efficiency

##Following losses have been found respectively
###Stator winding losses
###Rotor winding losses
###Slip ring brush losses
###Stator teeth fundamental losses

In [69]:
P_cos=3*R_s*rated_current*rated_current
print P_cos

5530.63584483


In [70]:
P_cor=3*R_r*rated_current*rated_current
print P_cor

4424.50867586


In [71]:
Vdrop=0.75
P_sr=3*Vdrop*rated_current_2
print P_sr

1249.62341815


In [72]:
G_ts=((3.14/4)*(math.pow(D_is+2*h_st*0.001,2)-math.pow(D_is,2))-h_st*0.001*slot_width*0.001*Ns)*7800*l_i
print G_ts

67.0021970651


import math
Piront=1.7*2.5*math.pow(f_1/50,1.3)*math.pow(1.32,1.7)*G_ts
print Piront

#####Stator yoke losses have been found with a similar approach

In [73]:
import math
G_ys=l_i*7800*3.14*(math.pow(D_out,2)-math.pow(D_is+2*h_st*0.001,2))/4
print G_ys

202.945161647


In [74]:
K_y=1.2
Pirony=1.2*2.5*math.pow(f_1/50.0,1.3)*math.pow(1.28,1.7)*G_ys
print Pirony

926.31117844


####now for the surface losses:

\begin{equation}
p_{siron}=5*10^5*B_{os}^2*\tau_r^2*K_0*(\frac{N_r*60*f_1}{p*10000})^{1.5}
\\P_{siron}=2*p*\tau*(\frac{\tau_s-b_{os}}{\tau_s})*l_i*K_{fe}*p_{siron}
\end{equation}

In [75]:
beta_r=0.41
beta_s=0.23
B_or=beta_r*1.48*B_g
B_os=beta_s*1.48*B_g
psiron=5*math.pow(10,5)*math.pow(B_os,2)*math.pow(rotor_tooth_pitch*0.001,2)*1.5*math.pow(Nr*60*f_1/(p*10000),1.5)
Psiron=2*2*pole_pitch*(slot_per_pole*1000-10)*l_i*0.95*psiron/(slot_per_pole*1000)
print psiron
print Psiron

442.96346151
75.7296801233


In [76]:
priron=5*math.pow(10,5)*math.pow(B_or,2)*math.pow(slot_per_pole,2)*1.5*math.pow(Nr*60*f_1/(p*10000),1.5)
Priron=2*p*pole_pitch*(rotor_tooth_pitch-5)*l_i*0.95*priron/(slot_per_pole*1000)
print Priron

250.551223911


#### for rotor flux pulsation losses:
\begin{equation}
\beta_{ps}= (K_{c2}-1)*B_g
\\\beta_{pr}= (K_{c1}-1)*B_g
\\P_{puls} = K_0*0.5*10^{-4}*((N_r*\frac{f}{p}*\beta_{ps})^2*G_{ts}+(N_s*\frac{f}{p}*\beta{pr})^2*G_{tr})
\end{equation}


In [77]:
K_0=2
beta_ps=(1.085-1)*B_g
beta_pr=(1.36-1)*B_g
G_tr=(3.14*((D_is-2*g)*(D_is-2*g)-(D_is-2*g-2*h_rt*0.001)*(D_is-2*g-2*h_rt*0.001))/4-Nr*h_rt*br*0.001*0.001)*l_i*7800
print G_tr

31.8980689198


In [78]:
P_puls=K_0*0.5*0.0001*((Nr*f_1*beta_ps/p)*(Nr*f_1*beta_ps/p)*G_ts+(Ns*f_1*beta_pr)*(Ns*f_1*beta_pr)*G_tr)
print P_puls

667.935231734


In [99]:
P_mec=0.01*150000
Ptot=P_n*1000+P_cos+P_cor+P_sr+Psiron+Priron+Pirony+P_puls+P_sr+P_mec
eff = P_n*1000/Ptot
print eff
Ploss=P_cos+P_cor+P_sr+Psiron+Priron+Pirony+P_puls+P_sr+P_mec

0.904295846543


####efficiency is %90 which is exactly the target value.

#Temperature rise
###temperature difference between the conductors in slots and the slot wall:
\begin{equation}
\Delta\theta_{co}=\frac{P_{co}}{\alpha_{cond}*A_{ls}}
\end{equation}


###frame temperature with respect to ambient air:
\begin{equation}
\Delta\theta_{frame}=\frac{\sum_{losses}}{\alpha_{conv}*A_{frame}}
\end{equation}

In [178]:
alpha_conv=350#cast iron frame contact heat coefficient for airgap 0.05-0.08mm#
alpha_cond=1.2/(slot_insulation*0.001)#if glass fibre is used#
print alpha_cond

300.0


In [179]:
A_lateral=(2*h_s+br)*0.001*L*Ns
print A_lateral

1.41020225526


In [180]:
A_frame=3.14*D_out*(L+pole_pitch)*3
print A_frame

2.64646914588


In [181]:
delta_conductors=P_cos/(alpha_cond*A_lateral)
delta_frame=Ploss/(alpha_conv*A_frame)
print delta_conductors
print delta_frame

13.0729140074
17.1386507792


In [185]:
ambient=40
winding_temperature=delta_frame+delta_conductors+ambient
print winding_temperature

70.2115647867
